### ISSUE TITLE
[META] What version of CommonMark does each release comply with?

### ISSUE THREAD

**Issue by K4LCIFER** [rlidwka/markdown-it.rs#45]
I wasn't able to find any reference to a specific version of CommonMark that this project is attempting to comply with.

**Comment by K4LCIFER** [comment]
A [search](https://github.com/search?q=repo%3Amarkdown-it-rust%2Fmarkdown-it+commonmark&type=code&p=1) for "commonmark" in this repository shows many references to 0.30 , but there are a few that show other versions, eg:
- https://github.com/markdown-it-rust/markdown-it/blob/2d7c085046a144d221490331b25ca565ecddbb1b/src/parser/inline/builtin/skip_text.rs#L52
- https://github.com/markdown-it-rust/markdown-it/blob/2d7c085046a144d221490331b25ca565ecddbb1b/src/plugins/cmark/block/reference.rs#L172

This is rather contradictory. At any rate, the current CommonMark spec is 0.31.2, so this package is not up to date with upstream.

**Comment by Steve0Greatness** [comment]
TL;DR: this crate, markdown-it.rs, aleardy seems to comply with version 0.31.2, even if incidentally.

This might present a problem if [0.31.2](https://spec.commonmark.org/0.31.2/changes.html) weren't just upkeep for the spec: updating outdated URLs, in some cases linking to archive.org archives, correcting spelling/grammar errors, etc.

The largest changes are a change of definition of punctuation marks in Unicode from "an ASCII punctuation character or anything in the general Unicode categories  `Pc`, `Pd`, `Pe`, `Pf`, `Pi`, `Po`, or `Ps`" _to_ "a character in the Unicode `P` (puncuation) or `S` (symbol) general categories."

In addition, there was the removal of a single hyphen to be interpreted as an empty list rather than setext heading (likely due to setext headers being able to be any length, including just 1); however, this crate already appears to comply with this (this can be verified on the [online demo](https://markdown-it-rust.github.io/markdown-it/)).
 
> * https://github.com/markdown-it-rust/markdown-it/blob/2d7c085046a144d221490331b25ca565ecddbb1b/src/parser/inline/builtin/skip_text.rs#L52
> * https://github.com/markdown-it-rust/markdown-it/blob/2d7c085046a144d221490331b25ca565ecddbb1b/src/plugins/cmark/block/reference.rs#L172

Second seems like The author stating that as of CommonMark 0.20, empty labels were disallowed, and not that the version as of now is 0.20. As for the first, I'm unsure--it could be updated, but given the only difference of that definition since 0.15 seems to be that at some point more explicit unicode character points were added, so this would likely be unnecessary.
